tests/unittests/tests-mtd_flashpage/tests-mtd_flashpage.c

```diff
@@ -0,0 +1,128 @@
+{
+    mtd_erase(dev, (FLASHPAGE_NUMOF - 1) * FLASHPAGE_SIZE, dev->pages_per_sector * dev->page_size);
+}
+
+static void test_mtd_init(void)
+{
+    int ret = mtd_init(dev);
+    TEST_ASSERT_EQUAL_INT(0, ret);
+}
+
+static void test_mtd_erase(void)
+{
+    /* Erase last sector */
+    int ret = mtd_erase(dev, (FLASHPAGE_NUMOF - 1) * FLASHPAGE_SIZE, FLASHPAGE_SIZE);
+    TEST_ASSERT_EQUAL_INT(0, ret);
+
+    /* Erase with wrong size (less than sector size) */
+    ret = mtd_erase(dev, (FLASHPAGE_NUMOF - 1) * FLASHPAGE_SIZE, dev->page_size);
+    TEST_ASSERT_EQUAL_INT(-EOVERFLOW, ret);
+
+    /* Unaligned erase */
+    ret = mtd_erase(dev, (FLASHPAGE_NUMOF - 1) * FLASHPAGE_SIZE + dev->page_size, dev->page_size);
+    TEST_ASSERT_EQUAL_INT(-EOVERFLOW, ret);
+
+    /* Erase 2 last sectors */
+    ret = mtd_erase(dev, (FLASHPAGE_NUMOF - 2) * FLASHPAGE_SIZE,
+                    FLASHPAGE_SIZE * 2);
+    TEST_ASSERT_EQUAL_INT(0, ret);
+
+    /* Erase out of memory area */
+    ret = mtd_erase(dev, (FLASHPAGE_NUMOF - 1) * FLASHPAGE_SIZE,
+                    FLASHPAGE_SIZE * 2);
+    TEST_ASSERT_EQUAL_INT(-EOVERFLOW, ret);
+}
+
+static void test_mtd_write_erase(void)
+{
+    const char buf[] = "ABCDEFGHIJK";
+    uint8_t buf_empty[] = {0xff, 0xff, 0xff};
+    char buf_read[sizeof(buf) + sizeof(buf_empty)];
+    memset(buf_read, 0, sizeof(buf_read));
+
+    int ret = mtd_write(dev, buf, (FLASHPAGE_NUMOF - 1) * FLASHPAGE_SIZE, sizeof(buf));
+    TEST_ASSERT_EQUAL_INT(sizeof(buf), ret);
+
+    ret = mtd_erase(dev, (FLASHPAGE_NUMOF - 1) * FLASHPAGE_SIZE, dev->pages_per_sector * dev->page_size);
+    TEST_ASSERT_EQUAL_INT(0, ret);
+
+    uint8_t expected[sizeof(buf_read)];
+    memset(expected, 0, sizeof(expected));
+    ret = mtd_read(dev, buf_read, (FLASHPAGE_NUMOF - 1) * FLASHPAGE_SIZE, sizeof(buf_read));
+    TEST_ASSERT_EQUAL_INT(sizeof(buf_read), ret);
+    TEST_ASSERT_EQUAL_INT(0, memcmp(expected, buf_read, sizeof(buf_read)));
+}
+
+static void test_mtd_write_read(void)
+{
+    const char buf[] = "ABCDEFG";
+    uint8_t buf_empty[] = {0, 0, 0, 0};
+    char buf_read[sizeof(buf) + sizeof(buf_empty)];
+    memset(buf_read, 0, sizeof(buf_read));
+
+    /* Basic write / read */
+    int ret = mtd_write(dev, buf, (FLASHPAGE_NUMOF - 1) * FLASHPAGE_SIZE, sizeof(buf));
+    TEST_ASSERT_EQUAL_INT(sizeof(buf), ret);
+
+    ret = mtd_read(dev, buf_read, (FLASHPAGE_NUMOF - 1) * FLASHPAGE_SIZE, sizeof(buf_read));
+    TEST_ASSERT_EQUAL_INT(sizeof(buf_read), ret);
+    TEST_ASSERT_EQUAL_INT(0, memcmp(buf, buf_read, sizeof(buf)));
+    TEST_ASSERT_EQUAL_INT(0, memcmp(buf_empty, buf_read + sizeof(buf), sizeof(buf_empty)));
+
+    ret = mtd_erase(dev, (FLASHPAGE_NUMOF - 1) * FLASHPAGE_SIZE, dev->pages_per_sector * dev->page_size);
+    TEST_ASSERT_EQUAL_INT(0, ret);
+
+    /* Unaligned write / read */
+    ret = mtd_write(dev, buf, (FLASHPAGE_NUMOF - 1) * FLASHPAGE_SIZE + sizeof(buf_empty), sizeof(buf));
+    TEST_ASSERT_EQUAL_INT(sizeof(buf), ret);
```

Comment:
Is this actually expected? I'm trying to fix this code since it doesn't work for several platforms but then I have some errors maybe related to the actual test.

I was expecting here a `-EINVAL`.